### PR TITLE
MULE-12799: Move ArtifactDeclaration to a new API

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>${slf4jVersion}</version>
+        </dependency>
+
+        <dependency>
+            <!-- TODO MULE-11007: when implemented, delete this dependency -->
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>${gsonVersion}</version>
```
Comment (gsfernandes) [inline pom.xml]: Do you need this dependency?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <!-- TODO MULE-11007: when implemented, delete this dependency -->
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>${commonsIoVersion}</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
```
Comment (gsfernandes) [inline pom.xml]: Same here, do a dependency:validate (or whatever the name of the plugin is) to validate if a dependency is not used here or used and not declared.

File: src/main/resources/META-INF/mule-module.properties [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.name=mule-api
```
Comment (gsfernandes) [inline src/main/resources/META-INF/mule-module.properties]: Change the name of the module. It is using mule-api which is wrong

File: src/main/resources/META-INF/mule-module.properties [hunk 4]
```diff
@@ -0,0 +1,5 @@
+module.name=mule-artifact-declaration
+
+artifact.export.classPackages=org.mule.runtime.app.declaration.api,\
+                              org.mule.runtime.app.declaration.api.fluent,\
+                              org.mule.runtime.app.declaration.internal.component.location
```
Comment (alegmarra) [inline src/main/resources/META-INF/mule-module.properties]: no export of internals

**Pull request merged**